File: coderadar-core/src/main/java/io/reflectoring/coderadar/analyzer/domain/Commit.java
package io.reflectoring.coderadar.analyzer.domain;

import lombok.Data;
import lombok.ToString;

import java.util.ArrayList;
import java.util.Date;
import java.util.LinkedList;
import java.util.List;

/** Metadata about a commit to a Git repository. */
@Data
@ToString(exclude = "touchedFiles")
public class Commit {
  private Long id;
  private String name;
  private Date timestamp;
  private String comment;
  private String author;
  private boolean merged = false;
  private boolean analyzed = false;

  private List<Commit> parents = new ArrayList<>();

  private List<FileToCommitRelationship> touchedFiles = new LinkedList<>();

  private List<MetricValue> metricValues = new LinkedList<>();

  public void setComment(String comment) {
    if (comment.length() > 255) {
      // truncating commit message if too long for database
      this.comment = comment.substring(0, 252) + "...";
    } else {
      this.comment = comment;
    }
  }
}

File: coderadar-ui/src/app/model/commit.ts
export class Commit {
  name: string;
  author: string;
  comment: string;
  timestamp: number;
  analyzed: boolean;

  constructor(commit?: any) {
    if (commit !== undefined && commit !== null) {
      this.name = commit.name;
      this.author = commit.author;
      this.comment = commit.comment;
      this.timestamp = commit.timestamp;
      this.analyzed = commit.analyzed;
    }
  }
}
